Userland/Libraries/LibJS/Runtime/ArrayBuffer.cpp

```diff
@@ -27,7 +32,13 @@ ArrayBuffer::ArrayBuffer(size_t byte_size, Object& prototype)
 
 ArrayBuffer::ArrayBuffer(ByteBuffer& buffer, Object& prototype)
     : Object(prototype)
-    , m_buffer(buffer)
+    , m_buffer(ByteBuffer { buffer })
+{
+}
+
+ArrayBuffer::ArrayBuffer(ByteBuffer* buffer, Object& prototype)
+    : Object(prototype)
+    , m_buffer(static_cast<ByteBuffer*>(buffer))
```

Comment:
`static_cast`'ing a `ByteBuffer*` to `ByteBuffer*` might not be needed either?

```suggestion
    , m_buffer(buffer)
```